src/ODEmodel.jl

```diff
@@ -74,21 +74,39 @@ function predict(p, g_0::Real, t, nG1::Integer, nG2::Integer, nD1, nD2)
     G1 = Vector{eltype(p)}(undef, length(t))
     G2 = Vector{eltype(p)}(undef, length(t))
 
-    for ii = 1:length(G1)
-        G1[ii] = sum(v[1:nG1]) + sum(v[(nG1 + nG2 + 1):(nG1 + nG2 + nD1)])
-        G2[ii] = sum(v[(nG1 + 1):(nG1 + nG2)]) + sum(v[(nG1 + nG2 + nD1 + 1):(nG1 + nG2 + nD1 + nD2)])
+    if nD1 == 0 & nD2 != 0
```

Comment:
Why do you need an if statement here? If nD1 is equal to 0 doesn't it just end up having no effect on addition?

Reply:
In either of G1 or G2, the second term `sum(...)` would be meaningless, for example in G1, since it is summing the index `nG1+nG2+1 : nG1+nG2` if nD1= 0.